### PR TITLE
BLO-84 initial implementation

### PR DESCRIPTION
**Pull request by saltz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,58 @@
...
+```ts
+DIContainer.bind<ObjectHasher>(ObjectHasher).toSelf().inTransientScope();
+```
+
+*consumer (typically a service)*
+```ts
+class MainService {
+    private objectHasher: ObjectHasher;
+
+    constructor(@inject(ObjectHasher) objectHasher: DataAccessLayer) {
```
Comment (TimoF97) [inline README.md]: I figure the `DataAccessLayer` is a copy & paste typo?

File: src/cryptoKeyUtil.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.ec = new EC("secp256k1");
+    }
+
+    /**
+     * Generate a new (EC) crypto key pair.
+     * @returns {EC.KeyPair | IKeyPair} EC key pair.
+     * @throws {KeyPairException} when the generated key pair fails to verify.
+     */
+    public generateKeyPair(): EC.KeyPair {
+        logger.info("Generating key pair.");
```
Comment (TimoF97) [inline src/cryptoKeyUtil.ts]: I personally think that if we're going with the approach of a separate crypto library, this library is not expected to perform any logging on its own. I would rather expect the consuming application to perform logging.

tdlr; should we perform logging within this library?

File: src/cryptoKeyUtil.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        const validationResult = keyPair.validate();
+
+        if (validationResult.result) {
+            logger.info("Generated key pair successfully.");
+
+            return keyPair;
+        }
+
+        throw new KeyPairException("Keypair generation failed", validationResult.reason);
```
Comment (TimoF97) [inline src/cryptoKeyUtil.ts]: The usage of "key pair / keypair" is inconsistent. Please stick to either "key pair" or "keypair" throughout the entire library, including the docs.

File: src/types/keyPair.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { ec as EC } from "elliptic";
+
+// tslint:disable-next-line: no-empty-interface
+export interface IKeyPair extends EC.KeyPair {
```
Comment (TimoF97) [inline src/types/keyPair.ts]: I probably missed the reasoning behind this empty interface while swiftly checking this PR; could you enlighten the need for the empty interface?

File: README.md (hunk 1 above)
Comment (saltz) [reply README.md]: Oeps it is.

File: src/types/keyPair.ts (hunk 4 above)
Comment (saltz) [reply src/types/keyPair.ts]: To expose the functions and properties of the parent interface: `EC.KeyPair` to the consumers. This was the quick and dirty way of achieving this.

File: src/cryptoKeyUtil.ts (hunk 3 above)
Comment (saltz) [reply src/cryptoKeyUtil.ts]: Yes it should be "key pair".

**Pull request merged**